webapp/src/js/controllers/reports.js

```diff
@@ -322,23 +334,28 @@ angular
       Modal({
         templateUrl: 'templates/modals/edit_report.html',
         controller: 'EditReportCtrl',
-        model: { report: $scope.selected[0].doc },
+        model: { report: ctrl.selected[0].doc },
       });
     });
 
     $scope.$on('VerifyReport', function(e, valid) {
-      if ($scope.selected[0].doc.form) {
+      if (ctrl.selected[0].doc.form) {
         $scope.setLoadingSubActionBar(true);
 
-        var doc = $scope.selected[0].doc;
-        if (doc.contact) {
-          doc.contact = lineage.minifyLineage(doc.contact);
+        if (ctrl.selected[0].doc.contact) {
+          var minifiedContact = lineage.minifyLineage(ctrl.selected[0].doc.contact);
+          ctrl.setFirstSelectedDocProperty({ contact: minifiedContact });
         }
 
-        doc.verified = doc.verified === valid ? undefined : valid;
+        var verified = ctrl.selected[0].doc.verified === valid ? undefined : valid;
+        ctrl.setFirstSelectedDocProperty({ verified: verified });
 
         DB()
-          .post(doc)
+          .get(ctrl.selected[0].doc._id)
+          .then(function(doc) {
+            ctrl.setFirstSelectedDocProperty({ _rev: doc._rev });
+            return DB().post(ctrl.selected[0].doc);
```

Comment:
I'm not sure how this was working before, but the doc was automatically getting updated with the new `rev`, perhaps as a side effect of `DB().post`? Anyway, as redux is entirely immutable the rev needs to be explicitly updated on the doc.

Reply:
Yeah almost certainly what you said. Mutable by default breeds funky behaviour..